libazureinit/src/media.rs

```diff
@@ -75,20 +77,27 @@ pub const PATH_MOUNT_DEVICE: &str = "/dev/sr0";
 pub const PATH_MOUNT_POINT: &str = "/run/azure-init/media/";
 
 const CDROM_VALID_FS: &[&str] = &["iso9660", "udf"];
+const MTAB_PATH: &str = "/etc/mtab";
 
 // Get a mounted device with any filesystem for CDROM
 #[instrument]
-pub fn get_mount_device() -> Result<Vec<String>, Error> {
-    let mut list_devices: Vec<String> = Vec::new();
+pub fn get_mount_device(path: Option<&Path>) -> Result<Vec<String>, Error> {
+    let fstab = FsTab::new(path.unwrap_or(Path::new(MTAB_PATH)));
+    let entries: Vec<FsEntry> = fstab.get_entries()?;
 
-    while let Some(device) = block_utils::get_mounted_devices()?
+    // Retrieve the names of all devices that have cdrom-type filesystem (e.g., udf)
+    let cdrom_devices: Vec<String> = entries
         .into_iter()
-        .find(|dev| CDROM_VALID_FS.contains(&dev.fs_type.to_str()))
-    {
-        list_devices.push(device.name);
-    }
-
-    Ok(list_devices)
+        .filter_map(|entry| {
+            if CDROM_VALID_FS.contains(&entry.vfs_type.as_str()) {
+                Some(entry.fs_spec.clone())
```

Comment:
You don't actually need the `clone()` here. The reason is that the list of entries is being consumed with `into_iter()` and giving us owned values. Since we don't hold onto the entry and reference it elsewhere, we are free to take ownership of the `fs_spec` it contains rather than create a new value from it.

Again, this is a micro-optimization that will never matter in practice, but just an FYI 😄